graphene_sqlalchemy/fields.py

```diff
@@ -81,8 +84,51 @@ def get_query(cls, model, info, sort=None, **args):
 
     @classmethod
     def resolve_connection(cls, connection_type, model, info, args, resolved):
+        session = get_session(info.context)
+        if resolved is None:
+            if not is_sqlalchemy_version_less_than("1.4") and isinstance(
```

Comment:
The problem here is the `isinstance` check. The import of it only happens if the version is nor less than 1.4. So in all other versions I need to fail the if statement before doing the `isinstance` check. I can imagine to ways around:
* do the startup check as you described and store the result in a constant `SQLALCHEMY_LESS_THAN_1_4`, so I just need to check the boolean value of the constant
* Invert this statement to check if it is a synchronous session

I would prefer the first option, as I am not sure how to check if a session is sync in sqlalchemy. There seems to be a protected class attribute `_is_asyncio` with sessions, but checking that would probably agin require checking the version of sqlalchemy first as I assume the property is new 😉

Reply:
Startup check option sounds great!

Reply:
We'll need this anyways as 2.0 is in beta; so we should use 1.4 style on all queries if possible - even in sync session. Scope of a different PR though. This will also require the global option.

Reply:
Implemented the startup option 👍